### DISCUSSION TITLE
[Android]How to render gltf without showing on the screen?

### DISCUSSION THREAD

**Discussion by duy-maimanh** [google/filament#4480] [Q&A] [answered]
I am developing an app that uses camera and filament on Android. I don't use TextureView or SurfaceView because I don't want to display it on the screen. So I use SurfaceTexture directly.
The idea is combine 2 SurfaceTexture (1 camera, 1 filament) together.
I've got a SurfaceTexture camera. I then tried to create a new SurfaceTexture for the filament but got an error. 
First I created a surface:
`val surfaceTexture = SurfaceTexture(191, false)`
`val surface = Surface(surfaceTexture)`
Then I created a swapchain:
`swapChain = engine.createSwapChain(surface)`
Error: `BufferQueueProducer: [SurfaceTexture-191-23376-0] dequeueBuffer: BufferQueue has been abandoned`
Do you have any suggestions?

**Comment by romainguy** [comment, marked as answer]
Where is that texture id from? You could also just render the camera stream with Filament directly.

**Reply by duy-maimanh** [reply]
> Where is that texture id from? 

I just create a SurfaceTexture with random texture id is 191.

> You could also just render the camera stream with Filament directly.

Do you mean to do the same sample-hello-camera ?

**Reply by romainguy** [reply]
Was the SurfaceTexture created with a proper GL context etc.? From the error message it doesn't look like it. Either that or your SurfaceTexture was garbage collected. Make sure you keep a reference to it.

Btw do you need a SurfaceTexture? You could also use an ImageReader depending on what you are doing.

And yes I meant like in the camera sample.

**Reply by duy-maimanh** [reply]
Because I use cameraX, I got a SurfaceTexture from cameraX PreviewOutput. Should I change to camera2 ?

**Reply by romainguy** [reply]
You can use the SurfaceTexture as a Filament texture. See the Stream API in Filament.
